litestar/plugins/problem_details.py

```diff
@@ -0,0 +1,120 @@
+
+if TYPE_CHECKING:
+    from litestar.config.app import AppConfig
+
+ProblemDetailsExceptionT = TypeVar("ProblemDetailsExceptionT", bound="ProblemDetailsException")
+ProblemDetailsExceptionHandlerType = Callable[[Request, ProblemDetailsExceptionT], Response]
+ExceptionToProblemDetailMapType = Mapping[type[ExceptionT], Callable[[ExceptionT], ProblemDetailsExceptionT]]
+
+
+def _problem_details_exception_handler(request: Request, exc: ProblemDetailsException) -> Response:
+    return exc.to_response(request)
+
+
+def _create_exception_handler(
+    exc_to_problem_details_exc_fn: Callable[[ExceptionT], ProblemDetailsException], exc_type: type[ExceptionT]
+) -> ExceptionHandler:
+    def _exception_handler(req: Request, exc: exc_type) -> Response:
+        problem_details_exc = exc_to_problem_details_exc_fn(exc)
+
+        return problem_details_exc.to_response(req)
+
+    return _exception_handler
+
+
+def _http_exception_to_problem_detail_exception(exc: HTTPException) -> ProblemDetailsException:
+    return ProblemDetailsException(status_code=exc.status_code, detail=exc.detail, extra=exc.extra, headers=exc.headers)
+
+
+class ProblemDetailsException(HTTPException):
+    """A problem details exception as per RFC 9457."""
+
+    _PROBLEM_DETAIL_MEDIA_TYPE = "application/problem+json"
+
+    def __init__(
+        self,
+        *args: Any,
+        detail: str = "",
+        status_code: int | None = None,
+        headers: dict[str, str] | None = None,
+        extra: dict[str, Any] | list[Any] | None = None,
+        type_: str = "about:blank",
+        title: str | None = None,
+        instance: str | None = None,
+    ) -> None:
+        """Initialize ``ProblemDetailsException``.
+
+        Args:
+            *args: if ``detail`` kwarg not provided, first arg should be error detail.
+            detail: Exception details or message. Will default to args[0] if not provided.
+            status_code: Exception HTTP status code.
+            headers: Headers to set on the response.
+            extra: An extra mapping to attach to the exception.
+            type_: The type field in the problem details.
+            title: The title field in the problem details.
+            instance: The instance field in the problem details.
+        """
+
+        super().__init__(*args, detail=detail, status_code=status_code, headers=headers, extra=extra)
+
+        self.type_ = type_
+        self.title = title
+        self.instance = instance
+
+    def to_response(self, request: Request) -> Response:
+        """Convert the problem details exception into a ``Response.``"""
+
+        problem_details = {"type": self.type_, "status": self.status_code}
+        for attr in ("title", "instance", "detail", "extra"):
+            if (value := getattr(self, attr, None)) is not None:
+                problem_details[attr] = value
+
+        return Response(problem_details, headers=self.headers, media_type=self._PROBLEM_DETAIL_MEDIA_TYPE)
```

Comment:
This Response should get the status code.

Reply:
Good catch!

Reply:
Is it possible to add/expand a test to catch this?

Reply:
I've modified the tests to check for the status on the responses as well.